### PR TITLE
1.4.0 Changelog

### PR DESCRIPTION
**Pull request by clairelin135**
Add changes since 1.3.14. 

Will add major changes section + migration guide in a follow up PR.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+
+- Fixed a bug that caused lazy `AutoMaterializePolicy`'s to not materialize missing assets.
+- [ui] Fixed an issue where global search and large DAGs were broken when using `--path-prefix`.
+- Schedule and sensor run submissions are now kept up to date with the current workspace, fixing an issue where a stale reference to a server would be used in some conditions.
+
+### Breaking Changes
+
+- Support for Python 3.7 has been dropped.
+- `build_asset_reconciliation_sensor` (Experimental) has been removed. It was deprecated in 1.3 in favor of `AutoMaterializePolicy`.
+- `asset_key(s)` properties on `AssetIn` and `AssetDefinition` have been removed in favor of `key(s)`.
```
Comment (sryza) [inline CHANGES.md:27]: @smackesey I think we should give more detail here to make sure it's clear that we're not breaking a stable API.

Comment (sryza) [reply CHANGES.md:27]: Same with root_input_manager below

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,54 @@
...
+- Duplicate partition keys passed to `StaticPartitionsDefinition` will now raise an error.
```
Comment (sryza) [inline CHANGES.md:28]: I think we should message this as a bugfix

File: CHANGES.md [hunk 3]
```diff
@@ -1,5 +1,54 @@
...
+- [dagster-airbyte] In line with the deprecation of `non_argument_deps` in favor of `deps`, `build_airbyte_assets` now accepts a `deps` parameter.
+- [dagstermill] In line with the deprecation of `non_argument_deps` in favor of `deps`, `define_dagstermill_asset` now accepts a `deps` parameter.
+
+### Bugfixes
+
+- Fixed a bug that caused lazy `AutoMaterializePolicy`'s to not materialize missing assets.
+- [ui] Fixed an issue where global search and large DAGs were broken when using `--path-prefix`.
+- Schedule and sensor run submissions are now kept up to date with the current workspace, fixing an issue where a stale reference to a server would be used in some conditions.
+
+### Breaking Changes
```
Comment (sryza) [inline CHANGES.md:23]: Heads up I believe there are a bunch more dbt breaking changes that we'll want to add here after we write the migration guide.

File: CHANGES.md [hunk 4]
```diff
@@ -1,5 +1,54 @@
...
+- [dagster-k8s] The Helm chart now sets readiness probes on user code deployment servers by default. These can be disabled with `dagster-user-deployments.deployments.[...].readinessProbe.enabled=false`.
+- We’ve deprecated the `non_argument_deps` parameter of `@asset` and `@multi_asset` in favor of a new `deps` parameter. The new parameter makes it clear that this is a first-class way of defining dependencies, makes code more concise, and accepts `AssetsDefinition` and `SourceAsset` objects, in addition to the `str`s and `AssetKey`s that the previous parameter accepted.
+- The `UPathIOManager` can now be extended to load multiple partitions asynchronously (Thanks Daniel Gafni!).
+- By default, Dagster will now automatically load default config values into the launchpad. This behavior can be disabled in the user settings page.
+- [dagster-airbyte] In line with the deprecation of `non_argument_deps` in favor of `deps`, `build_airbyte_assets` now accepts a `deps` parameter.
+- [dagstermill] In line with the deprecation of `non_argument_deps` in favor of `deps`, `define_dagstermill_asset` now accepts a `deps` parameter.
```
Comment (alangenfeld) [inline CHANGES.md:15]: nit: reorder the `[library]` entries below the non lib entries

**Review by alangenfeld — APPROVED** [review]

File: CHANGES.md (hunk 2 above)
Comment (smackesey) [reply CHANGES.md:28]: It was previously raising a deprecation warning.

**Review by sryza — APPROVED** [review]

**Pull request merged**